sw/device/lib/testing/hmac_testutils.c

```diff
@@ -8,6 +8,54 @@
 #include "sw/device/lib/runtime/ibex.h"
 #include "sw/device/lib/testing/test_framework/check.h"
 
+const char kRefData[34] = "Sample message for keylen=blocklen";
+
+const uint8_t kRefHmacLongKey[100] = {
```

Comment:
I think here might be some naming convention misalignment :)
In our OT HMAC, keys are always 256 bits (https://docs.opentitan.org/hw/ip/hmac/doc/index.html#Reg_key_0), but message can be random length.
I think here since you are using:
```
  hmac_testutils_push_message(&hmac, (char *)kRefHmacLongKey,
                              sizeof(kRefHmacLongKey));
```
to write the data into fifo, in OT hmac, we call them message or input data instead of keys.
Keys are 256 bits and only used in hmac mode, not used in sha256 mode :)

Reply:
I hope the added comments in clkmgr_testutils.h explaining how the reference data is used clarify this issue.